### ISSUE TITLE
Error in calculation of grid average inside density_from_Universe (version 0.13.0)

### ISSUE THREAD

**Issue by Balasubra** [MDAnalysis/mdanalysis#716]
Inside density.py script, under density_from_Universe section, the average histogram is obtained as follows (lines 515-518), 
      h[:], edges[:] = np.histogramdd(coord, bins=bins, range=arange, normed=False)
      grid += h  # accumulate average histogram
n_frames = u.trajectory.n_frames
grid /= float(n_frames)

If the user choose to use trajectory indexing, the grid average may be wrong, instead the code may go like this,
      h[:], edges[:] = np.histogramdd(coord, bins=bins, range=arange, normed=False)
      grid += h  # accumulate average histogram
n_frames = (stop - start)/step  # Instead average is done over the analysed portion
grid /= float(n_frames)

**Comment by jbarnoud** [comment]
For reference, the quoted code starts at [line 515 of `analysis/density.py`.](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/density.py#L515)

The remark make sense. Please go ahead with a pull request :smile:. Some small comments, though:

* in python 3, the `/` operator makes a true division rather than an integer one; here the `//` operator should be used
* `start`, `stop`, and `step` can each be `None`, which would make the expression fail; maybe we could have a function in the Reader that calculates the number of frames from `start`, `stop`, and `step`, accounting for `None` values?

**Comment by richardjgowers** [comment]
@jbarnoud `Reader.check_slice_indices` does exactly this

Doing:
``` python
from six.moves import range

start, stop, step = u.trajectory.check_slice_indices(start, stop, step)

nframes = len(range(start, stop, step))
```

Should do the trick

**Comment by jbarnoud** [comment]
@richardjgowers Nice! I love when everything is already there!

**Comment by Balasubra** [comment]
How do i make a copy of the recent develop branch and do the changes and pull request.

**Comment by Balasubra** [comment]
I hope i have done it correctly.

**Comment by jbarnoud** [comment]
I reopen the issue until the PR get merged.

**Comment by orbeckst** [comment]
@jbarnoud thanks for re-opening.

@Balasubra , don't worry about closing issues, typically one of the developers takes care of it or it happens automatically when your PR gets merged (add a line to commit message such as "closes #716"). We want to keep issues open until the problem is fixed in develop or we decided not to take action or if it was resolved by some other means.

**Comment by orbeckst** [comment]
This was closed with merge of PR #718.

(If you add to the commit message a line
```
- closes #716
```

then it will automatically close the associated issue and link itself.

**Comment by Balasubra** [comment]
- closes #716
